app/components/subcase/description-panel/edit.js

```diff
@@ -122,13 +122,31 @@ export default class SubcaseDescriptionEdit extends Component {
     if (this.agendaItemType.uri !== oldAgendaItemType.uri) {
       await this.updateNewsletterAfterRemarkChange();
       await this.updateDecisionReports();
+      await this.recalculateAgendaitemNumber();
```

Comment:
Recalculating after the agendaitem has been saved won't always work.

For example: this subcase is a nota with number 7. There are 3 mededelingen on the agenda. When switching the type to mededeling, the query in this function (after fixing it) will find this subcase (as 7 is larger than 3) and set its number to 8.

We need to perform this calculation before saving, or at the very least filter out the current agendaitem when trying to find the number.